### ISSUE TITLE
[Feature] Consider realtime-piping webServer logs

### ISSUE THREAD

**Issue by stdedos** [microsoft/playwright#23313]
Improve https://github.com/microsoft/playwright/issues/22454 by allowing to also stream the webServer output in "realtime".

Waiting on a timeout (of 60000ms) to show that the webServer failed is kinda sad.

There are workarounds (`DEBUG=` and lower timeout), but they have their own drawbacks.

**Comment by aslushnikov** [comment]
@stdedos what is the 60000ms timeout you're mentioning? I run the following and it works well for me:

```ts
// server.js
setInterval(() => console.log('Time: ', Date.now()), 100);

// Create a server instance
require('http').createServer((req, res) => {
  res.writeHead(200, { 'Content-Type': 'text/plain' });
  res.end('Hello, World!');
}).listen(1234, () => {
  console.log('Server is running on port 1234');
});
```

```ts
// playwright.config.ts
import { defineConfig } from '@playwright/test';

export default defineConfig({
  webServer: {
    command: 'node ./server.js',
    port: 1234,
    stdout: 'pipe',
  },
});
```

```ts
// a.spec.ts
import { test } from '@playwright/test';

test('should work', async ({ }) => {
  await new Promise(x => setTimeout(x, 1000));
});
```

Running this yields immediate results:

```sh
aslushnikov:~/tmp$ npx playwright test

Running 1 test using 1 worker

[WebServer] Server is running on port 1234
[WebServer] Time:  1685138975057
[WebServer] Time:  1685138975159
[WebServer] Time:  1685138975259
  ✓  1 a.spec.ts:3:5 › should work (1.0s)
[WebServer] Time:  1685138975360
[WebServer] Time:  1685138975461
[WebServer] Time:  1685138975562
[WebServer] Time:  1685138975663
[WebServer] Time:  1685138975765
[WebServer] Time:  1685138975866
[WebServer] Time:  1685138975967
[WebServer] Time:  1685138976067
[WebServer] Time:  1685138976169
[WebServer] Time:  1685138976269

  1 passed (1.2s)
aslushnikov:~/tmp$
```

**Comment by stdedos** [comment]
In my case, `webServer` is `npm run dev`, which in turn is `nodemon`, which gives me:

```
[WebServer] [nodemon] 2.0.22
[WebServer] [nodemon] to restart at any time, enter `rs`
[WebServer] [nodemon] watching path(s): src/**/*
[WebServer] [nodemon] watching extensions: ts
[WebServer] [nodemon] starting `ts-node -r ./src/initialize-telemetry.ts --transpile-only ./src/index.ts`
[WebServer] {
  "errors": [
    {
      "error": "undefined",
      "variable": "CONFIGURATION_STORAGE_CONTAINER_NAME"
    }
  ],
  "level": "error",
  "message": "Error starting application Env vars are missing or invalid",
  "name": "EnvVarError",
  "stack": "...."
}
[WebServer] [nodemon] clean exit - waiting for changes before restart
```

... but only after the timeout expires 😅

**Comment by aslushnikov** [comment]
@stdedos so could you please build a small repo with all the right dependencies and scripts that illustrates the issue?

**Comment by stdedos** [comment]
I suck at building ts repos 😅 I only touch them 🤣

**Comment by aslushnikov** [comment]
Too bad! This is not actionable without a reproduction that we can run locally to see what's going on.

I'll have to close this; but feel free to file a new issue if you still can make a repro!

**Comment by andrewmclagan** [comment]
I can confirm this. I will attempt to replicate.

**Comment by andrewmclagan** [comment]
Here is the repository replicating this issue:

https://github.com/andrewmclagan/playwright-std-out

```
pnpm test:e2e
```

You will notice output is only displayed _**after**_ the process has completed and is not streamed/piped to `stdout` as per the configuration setting. This makes it very difficult to debug.

**Comment by mxschmitt** [comment]
Saw you created https://github.com/microsoft/playwright/issues/23539 so we continue there.

**Comment by stdedos** [comment]
_Please close as incomplete at least_ 🙏

**Comment by mxschmitt** [comment]
there is unfortunately only completed and no planned :/

sorry for the confusion! By default its completed.

**Comment by andrewmclagan** [comment]
All good, let me know if you need more information on the reproduction repo.

**Comment by raythurnevoid** [comment]
@andrewmclagan try to use `reuseExistingServer` option. I wanted to see the server logs in realtime even after the startup but seems like it's not possible within playwright. Using `stdout: 'pipe'` only pipes until the server starts, everything else is blackboxed.

**Comment by adueck** [comment]
I'm still running into this issue. I can't get the `stdout` from my webserver.  Here's a [repo example](https://github.com/adueck/haskell-mailer/tree/8eff8ef0b3c9c6a0131083d10386c7ccb62350c6)
 Nothing comes out from the `stdout` from my webserver. The only `stdout` that shows up is when I run an `execSync` to call some command in the test.
